### PR TITLE
[FISH-1019] GlassFishProperties throws NPE if the Properties is null

### PR DESCRIPTION
**Pull request by bhanuunrivalled**
GlassFishProperties throws NPE if the Properties is null

## Description
Doing a null Check and updated the JAVA DOC

### CODE REVIEW

**Comment by rdebusscher** [comment]
Hi @bhanuunrivalled ,

Thank you for the PR but before we can accept this, you need to sign the CLA and send it to us.

https://github.com/payara/Payara/blob/master/PayaraCLA.pdf

Thanks
Best Regards
Rudy

**Comment by bhanuunrivalled** [comment]
Thanks @rdebusscher 

attaching the CLA. 

[PayaraCLA-1.pdf](https://github.com/payara/Payara/files/5320103/PayaraCLA-1.pdf)

**Comment by bhanuunrivalled** [comment]
hi @rdebusscher ,

can I work on this ?

https://github.com/payara/Payara/issues/4795

please let me know your opinion . I have good knowledge about JSF.

Thanks.

**Comment by rdebusscher** [comment]
Deleted the previous comment as it contained personal information. The CLA is sent to the correct email address.

**Comment by AlanRoth** [comment]
jenkins test please

File: nucleus/common/simple-glassfish-api/src/main/java/org/glassfish/embeddable/GlassFishProperties.java [hunk 1]
```diff
@@ -74,8 +74,12 @@ public GlassFishProperties() {
      * for the plugged in {@link GlassFishRuntime} (if any)
      *
      * @param props Properties object which will back this GlassFishProperties object.
+     * @throws IllegalArgumentException if props is null
```
Comment (lprimak) [inline nucleus/common/simple-glassfish-api/src/main/java/org/glassfish/embeddable/GlassFishProperties.java:77]: `@throws NullPointerException`

File: nucleus/common/simple-glassfish-api/src/main/java/org/glassfish/embeddable/GlassFishProperties.java [hunk 2]
```diff
@@ -74,8 +74,12 @@ public GlassFishProperties() {
...
      */
     public GlassFishProperties(Properties props) {
+        if(props == null){
```
Comment (lprimak) [inline nucleus/common/simple-glassfish-api/src/main/java/org/glassfish/embeddable/GlassFishProperties.java:80]: this should be `Objects.requireNonNull(props, "props should not be null")`

Comment (lprimak) [reply nucleus/common/simple-glassfish-api/src/main/java/org/glassfish/embeddable/GlassFishProperties.java:80]: or something similar to that, introducing a different exception type might break some functionality and introduces unnecessary risks

**Review by lprimak — CHANGES_REQUESTED** [review]

**Comment by lprimak** [comment]
Superceded / incorporated into https://github.com/payara/Payara/pull/5101

**Comment by lprimak** [comment]
@bhanuunrivalled I could not push my changes on top of your PR so I created my own

If you can give me permissions I can reopen this and finish it here

**Comment by bhanuunrivalled** [comment]
> @bhanuunrivalled I could not push my changes on top of your PR so I created my own
> 
> If you can give me permissions I can reopen this and finish it here

OK @lprimak I give permission

**Comment by lprimak** [comment]
Sorry, still getting permissino denied
```
remote: Resolving deltas: 100% (16/16), completed with 10 local objects.
To github.com:bhanuunrivalled/Payara.git
 ! [remote rejected]       bhan-fix-3 -> master (permission denied)
error: failed to push some refs to 'git@github.com:bhanuunrivalled/Payara.git'

```

**Comment by lprimak** [comment]
Was able to push 10 minutes ago but not anymore

**Pull request closed without merging**